enos/enos-scenario-upgrade.hcl

```diff
@@ -698,6 +698,31 @@ scenario "upgrade" {
     }
   }
 
+  step "verify_billing_start_date" {
+    description = global.description.verify_billing_start_date
+    skip_step   = matrix.edition == "ce"
+    module      = module.vault_verify_billing_start_date
+    depends_on = [
+      step.get_updated_vault_cluster_ips,
+      step.verify_vault_unsealed,
+      step.verify_read_test_data,
+    ]
+
+    providers = {
+      enos = local.enos_provider[matrix.distro]
+    }
+
+    verifies = [
+      quality.vault_billing_start_date,
+    ]
+
+    variables {
+      vault_install_dir = global.vault_install_dir[matrix.artifact_type]
+      vault_instances   = step.create_vault_cluster_targets.hosts
```

Comment:
Nit: mind if we rename `vault_instances` this to `hosts` to align with most other modules?